streaming/stream_transfer_task.cc

```diff
@@ -237,18 +237,23 @@ future<> stream_transfer_task::execute() {
     }).then([this, id, plan_id] {
         _mutation_done_sent = true;
         sslog.debug("[Stream #{}] GOT STREAM_MUTATION_DONE Reply from {}", plan_id, id.addr);
-    }).handle_exception([this, plan_id, cf_id, id] (std::exception_ptr ep) {
+    }).handle_exception_type([this, plan_id, cf_id, id] (const replica::no_such_column_family& e) {
```

Comment:
This is changing the way we handle errors. Before this change, the exception could other exceptions other than no_such_column_family, e.g., the peer node found the table is dropped and closed the stream. In this case, even if it is no_such_column_family, we should still check if the table is gone and ignore the exception.

With this patch, it will throw error instead which is not what we want.

Reply:
> This is changing the way we handle errors.
> Before this change, the exception could other exceptions other than no_such_column_family, e.g., the peer node found the table is dropped and closed the stream. In this case, even if it is no_such_column_family, we should still check if the table is gone and ignore the exception.
> 
> With this patch, it will throw error instead which is not what we want.

i am not sure about this. with this change, in the branch of `handle_exception_type()`, we either return a `make_ready_future()` or return `send_stream_mutation_done(...)`, am i missing anything?

Reply:
E.g., send_mutation_fragments() throws rpc::closed_error because peer closed the rpc stream and the table is dropped. In this case, we should ignore the rcp::closed_erorr exception.

Reply:
i see. no wonder we are catching `std::exception_ptr` here. i will revise this change to query all shards as you suggest. didn't really want to go this way, as i thought it was overkill.